File: redsheet-portal/src/app/database-models/communication-plan.ts
import { required, maxLength, range, nested } from '@rx/annotations';
import { ProjectPreparation, vCommunicationPlanRecord } from './'
export class CommunicationPlan {
    constructor(communicationPlan?: CommunicationPlan | vCommunicationPlanRecord) {
        let properties = ["communicationPlanId", "mediaMeans", "message", "to", "projectPreparationId",];
        for (let property of properties)
            if (communicationPlan && communicationPlan[property])
                this[property] = communicationPlan[property];
    }

    communicationPlanId: number = 0;

    @required()
    @maxLength(100)
    mediaMeans: string = undefined;

    @required()
    @maxLength(200)
    message: string = undefined;

    @required()
    @maxLength(50)
    to: string = undefined;

    @range(0, 2147483647)
    projectPreparationId: number = undefined;
    projectPreparation: ProjectPreparation;


}

File: RedSheet.DbEntities/Models/CommunicationPlan.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
    [Table("CommunicationPlans", Schema = "dbo")]
    [Serializable]
    public partial class CommunicationPlan
    {

        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        [Key]
        public int CommunicationPlanId { get; set; }

        [Required]
        [MaxLength(100)]
        public string MediaMeans { get; set; }

        [Required]
        [MaxLength(200)]
        public string Message { get; set; }

        [Required]
        [MaxLength(50)]
        public string To { get; set; }

        [RelationshipTableAttribue("ProjectPreparations", "dbo")]
        //Relationships
        public int ProjectPreparationId { get; set; }

        [ForeignKey("ProjectPreparationId")]
        public ProjectPreparation ProjectPreparation { get; set; }

        public CommunicationPlan()
        {
        }

    }
}

